src/test/java/se/liu/ida/hefquin/queryplan/executable/impl/ops/ExecOpSymmetricHashJoinTest.java

```diff
@@ -0,0 +1,207 @@
+package se.liu.ida.hefquin.queryplan.executable.impl.ops;
+
+import org.apache.jena.graph.NodeFactory;
+import org.apache.jena.sparql.core.Var;
+import org.apache.jena.sparql.engine.binding.Binding;
+import org.junit.Test;
+import se.liu.ida.hefquin.data.SolutionMapping;
+import se.liu.ida.hefquin.data.jenaimpl.JenaBasedSolutionMapping;
+import se.liu.ida.hefquin.data.jenaimpl.JenaBasedSolutionMappingUtils;
+import se.liu.ida.hefquin.queryplan.executable.impl.GenericIntermediateResultBlockImpl;
+import se.liu.ida.hefquin.queryplan.executable.impl.MaterializingIntermediateResultElementSink;
+
+import java.util.Iterator;
+
+import static org.junit.Assert.*;
+
+public class ExecOpSymmetricHashJoinTest extends ExecOpTestBase
+{
+	@Test
+	public void JoinWithOneJoinVariable(){
+		final Var var1 = Var.alloc("v1");
+		final Var var2 = Var.alloc("v2");
+		final Var var3 = Var.alloc("v3");
+
+		final GenericIntermediateResultBlockImpl input1 = new GenericIntermediateResultBlockImpl();
+		input1.add( JenaBasedSolutionMappingUtils.createJenaBasedSolutionMapping(
+				var1, NodeFactory.createURI("http://example.org/x1"),
+				var2, NodeFactory.createURI("http://example.org/y1")) );
+		input1.add( JenaBasedSolutionMappingUtils.createJenaBasedSolutionMapping(
+				var1, NodeFactory.createURI("http://example.org/x2"),
+				var2, NodeFactory.createURI("http://example.org/y2")) );
+
+		final GenericIntermediateResultBlockImpl input2 = new GenericIntermediateResultBlockImpl();
+		input2.add( JenaBasedSolutionMappingUtils.createJenaBasedSolutionMapping(
+				var2, NodeFactory.createURI("http://example.org/y1"),
+				var3, NodeFactory.createURI("http://example.org/z1")) );
+		input2.add( JenaBasedSolutionMappingUtils.createJenaBasedSolutionMapping(
+				var2, NodeFactory.createURI("http://example.org/y1"),
+				var3, NodeFactory.createURI("http://example.org/z2")) );
+		input2.add( JenaBasedSolutionMappingUtils.createJenaBasedSolutionMapping(
+				var2, NodeFactory.createURI("http://example.org/y2"),
+				var3, NodeFactory.createURI("http://example.org/z3")) );
+
+		final MaterializingIntermediateResultElementSink sink = new MaterializingIntermediateResultElementSink();
+
+		final ExecOpSymmetricHashJoin op = new ExecOpSymmetricHashJoin();
+		op.preprocess(input1, input2, sink);
```

Comment:
This method is protected. You shouldn't even be able to get this line compiled. 

Moreover, even if this method was public, it would never be called during normal processing. During normal processing, only the four methods of the BinaryExecutionOp interface are called.